sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/arm/collection/implementation/BatchDeletionImpl.java

```diff
@@ -17,21 +16,17 @@
  */
 public class BatchDeletionImpl {
 
-    @SuppressWarnings({"unchecked", "rawtypes"})
     public static Flux<String> deleteByIdsAsync(Collection<String> ids,
                                                 BiFunction<String, String, Mono<Void>> deleteByIdAsync) {
         if (ids == null || ids.isEmpty()) {
             return Flux.empty();
+        } else {
+            return Flux.fromIterable(ids)
+                .flatMapDelayError(id -> {
+                    final String resourceGroupName = ResourceUtils.groupFromResourceId(id);
+                    final String name = ResourceUtils.nameFromResourceId(id);
+                    return ReactorMapper.map(deleteByIdAsync.apply(resourceGroupName, name), id);
```

Comment:
I think the deleteByIdAsync should take one string `id` rather than two strings (some resource will need more).

Reply:
We can easily overload another method with same name, and take `Function<String, Mono<Void>>` for this case (I actually wrote this signature first, then changed it since most case it is resourceGroupName+name).